aqueduct_client/aqueduct_client.py

```diff
@@ -0,0 +1,243 @@
+import datetime
+
+import requests
+import pandas as pd
+
+from .utils import load_api_key
+
+
+class AqueductClient(object):
+    def __init__(self, api_key=None, base_url=None):
+        """
+        Creates an AqueductClient.
+
+        Parameters
+        ----------
+        api_key : str, optional
+            The Quantopian API key to use.  If not given, we attempt
+            to load the key from a credentials file or from an
+            environment variable.
+
+        base_url : str, optional
+            The base URL for the Aqueduct API.  Defaults to the
+            FactSet Aqueduct endpoint.
+        """
+        if base_url:
+            self._base_url = base_url
+        else:
+            self._base_url = \
+                "https://factset.quantopian.com/api/experimental/pipelines"
+
+        if api_key is not None:
+            self._api_key = api_key
+        else:
+            self._api_key = load_api_key()
+
+    def get_all_pipelines(self):
+        """
+        Returns the metadata of all the pipelines you've run.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        List
+            A list of all the pipelines you have run.  Each pipeline is a dict
+            with id, start_date, end_date, created_at, status, and other
+            properties.
+        """
+        response = self._get('', None)
+        response.raise_for_status()
+        pipelines = response.json()['pipelines']
+        return pipelines
+
+    def get_pipeline(self, id):
+        """
+        Returns the metadata of a single pipeline.
+
+        Parameters
+        ----------
+        id : str
+            The id of the pipeline to load.
+
+        Returns
+        -------
+        dict
+            The metadata of the pipeline, containing id, start_date,
+            end_date, created_at, status, and other properties.
+        """
+        response = self._get('/{id}'.format(id=id), None)
+        response.raise_for_status()
+        pipeline = response.json()['pipeline']
+        return pipeline
+
+    def create_new_pipeline(self,
+                            code,
+                            start_date,
+                            end_date,
+                            name=None,
+                            params={},
```

Comment:
We shouldn't use a [mutable default argument](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments) here.